### PR TITLE
Add hash for test data

### PR DESCRIPTION
**Pull request by sven-oly**
There's now a key 'hexhash' in the test data, which is computed from the stringified version of the test withouth the 'label' key. The intent is to use this to compare the same test conditions across different ICU/CLDR versions.

This will also be used in handling "known issues" as a class of results.

### CODE REVIEW

File: testgen/generators/base.py [hunk 1]
```diff
@@ -20,7 +20,39 @@ def __init__(self, icu_version, run_limit=None):
...
+
+        for test in all_tests:
+            try:
+                test_no_label = test.copy()
+            except BaseException as error:
+                logging.error('error: %s, Item with no label found here: %s, %s' ,
+                              error, testdata['test_type'], test)
+                continue
+            del test_no_label['label']
+            test_no_label_string = json.dumps(test_no_label)
```
Comment (sffc) [inline testgen/generators/base.py:44]: Suggestion: to make this more stable, please remove empty fields and sort them:

- https://stackoverflow.com/a/60124334/1407170
- https://stackoverflow.com/a/23820416/1407170

Something similar can be done in JSON.

File: schema/lang_names/result_schema.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
+{"$schema": "https://json-schema.org/draft/2020-12/schema",
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply schema/lang_names/result_schema.json:1]: I'll resolve this with language_names.

File: schema/rdt_fmt/test_schema.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
+{"$schema": "https://json-schema.org/draft/2020-12/schema",
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply schema/rdt_fmt/test_schema.json:1]: No, it wasn't there before.

File: schema/lang_names/result_schema.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (sven-oly) [reply schema/lang_names/result_schema.json:1]: Resolving this in favor of "lang_names" everywhere.

File: testgen/generators/base.py (hunk 1 above)
Comment (sven-oly) [reply testgen/generators/base.py]: Thanks for helping address these last week. New commits fix this.

**Review by sffc — APPROVED** [review]

**Pull request merged**